pdr_backend/lake/table_bronze_pdr_slots.py

```diff
@@ -0,0 +1,203 @@
+        2. Transform slots to bronze
+        3. Concat to existing table
+    """
+
+    # only add new slots
+    slots_df = dfs["pdr_slots"].filter(
+        (pl.col("timestamp") >= ppss.lake_ss.st_timestamp)
+        & (pl.col("timestamp") <= ppss.lake_ss.fin_timestamp)
+        & (pl.col("ID").is_in(collision_ids).not_())
+    )
+
+    # select only the columns that we need
+    columns_to_select = [
+        "ID",
+        "slot",
+        "timestamp",
+        "roundSumStakesUp",
+        "roundSumStakes",
+    ]
+    slots_df = slots_df.select(columns_to_select)
+
+    # create contract column from the ID column
+    slots_df = slots_df.with_columns(
+        contract=pl.col("ID").map_elements(lambda s: s.split("-")[0])
+    )
+
+    if len(slots_df) == 0:
+        return dfs
+
+    if dfs[bronze_pdr_slots_table_name].schema == bronze_pdr_slots_schema:
+        dfs[bronze_pdr_slots_table_name] = dfs[bronze_pdr_slots_table_name].select(
+            slots_df.schema
+        )
+
+    # append to existing dataframe
+    new_bronze_df = pl.concat([dfs[bronze_pdr_slots_table_name], slots_df])
+    dfs[bronze_pdr_slots_table_name] = new_bronze_df
+    return dfs
+
+
+def _process_truevals(
+    dfs: Dict[str, pl.DataFrame], ppss: PPSS
+) -> Dict[str, pl.DataFrame]:
+    """
+    Perform post-fetch processing on the data
+    """
+    # get truevals within the update
+    truevals_df, _ = pick_df_and_ids_on_period(
+        target=dfs["pdr_truevals"],
+        start_timestamp=ppss.lake_ss.st_timestamp,
+        finish_timestamp=ppss.lake_ss.fin_timestamp,
+    )
+
+    # get ref to bronze_slots
+    slots_df = dfs[bronze_pdr_slots_table_name]
+
+    # add columns that are going to be populated
+    slots_df = slots_df.with_columns(last_event_timestamp=pl.lit(None))
+    slots_df = slots_df.with_columns(trueval=pl.lit(None))
+
+    initial_schema_keys = slots_df.schema.keys()
+
+    # update only the ones within this time range
+    slots_df = (
+        slots_df.join(truevals_df, left_on="ID", right_on="ID", how="left")
+        .with_columns(
+            [
+                pl.col("trueval").fill_null(pl.col("trueval")),
+                pl.col("timestamp_right").fill_null(pl.col("last_event_timestamp")),
+            ]
+        )
+        .drop(["trueval", "last_event_timestamp"])
+        .rename(
+            {
+                "trueval_right": "trueval",
+                "timestamp_right": "last_event_timestamp",
+            }
+        )
+        .select(initial_schema_keys)
+    )
+
+    # update dfs
+    dfs[bronze_pdr_slots_table_name] = slots_df
+    return dfs
+
+
+def _process_predictions(
+    dfs: Dict[str, pl.DataFrame], ppss: PPSS
+) -> Dict[str, pl.DataFrame]:
+    """
+    @description
+        Perform post-fetch processing on the data
+        1. Find predictions( within the update
+
+    """
+    # get predictions within the update
+    predictions_df, _ = pick_df_and_ids_on_period(
+        target=dfs["pdr_predictions"],
```

Comment:
We need to join with the bronze_pdr_predictions table.

**The raw pdr_predictions table should be generally avoided to be queried directly.**

Reply:
Isn't this going to make the two bronze tables too dependent from each other? This way we have to always make sure that the bronze predictions table is always up to date before being able to get up to date from the slots data

Reply:
1. Like we discussed, the `bronze_predictions` table should always be updated first.
2. `bronze_slots` should use `bronze_predictions` table in case there is any transformation/cleaning up that `bronze_predictions` has done.

Resolving this conversation.